react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+export default class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+
+  handleChange(e) {
+    this.setState({
+      password: e.target.value
+    });
+  }
+
+  handleSubmit(event) {
+
+    event.preventDefault();
+  }
+
+  render() {
+    let errText;
+    let iconClassName;
+    const usersInput = this.state.password;
+    if (usersInput.length === 0) {
+      iconClassName = 'fa-xmark';
+      errText = 'A passowrd is required';
+    } else if (usersInput.length < 8) {
+      iconClassName = 'fa-xmark';
+      errText = 'Your password is too short';
+    } else if (!usersInput.includes('!' || '@' || '#' || '$' || '%' || '&' || '*' || '(' || ')')) {
```

Comment:
this logic does not check the way you intend, it only checks for `'!'`.

You can check for all of these though

```js
const special = ['!', '@', ...];
if (!special.some(character => usersInput.includes(character)) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
https://user-images.githubusercontent.com/106217355/188209538-bc3eeaec-8dbd-4f20-8125-9d1cfa152fd8.mp4